enterprise/tomcat5/src/org/netbeans/modules/tomcat5/deploy/TomcatManagerImpl.java

```diff
@@ -333,7 +333,7 @@ private static String encodePath(String str) {
             }
             StringBuilder result = new StringBuilder();
             while (st.hasMoreTokens()) {
-                result.append("/").append(URLEncoder.encode(st.nextToken(), StandardCharsets.UTF_8)); // NOI18N
+                result.append("/").append(URLEncoder.encode(st.nextToken(), "UTF-8")); // NOI18N
```

Comment:
I am sometimes using `StandardCharsets.UTF_8.name()` to make it explicit that the exception can't occur. But this is ok too of course since everyone knows that UTF is standard.

Maybe instead of catching `Exception` and rethrowing it as RT ex, it would be better to catch `UnsupportedEncodingException` a few lines down?

Reply:
Valid points - pushed an update. Will merge, once green.